Add documentation for LG ThinQ integration (#34169)
home-assistant/home-assistant.github.io: source/_integrations/lg_thinq.markdown

- [Water Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-purifier/)
- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/wine-cellar/)
{% note %}
Air Purifier Fan, Washtower Dryer, Washtower Washer, Humidifier, Washcombo Main and Washcombo Mini are also supported.
{% endnote %}

## Troubleshooting

### Setup

#### Aborted: The token is not valid

This error occurs when the Personal Access Token (PAT) is invalid or entered incorrectly. Please visit the [Personal Access Token Page](https://d1jykc6oogauei.cloudfront.net/) page to check if your token is valid.

#### Aborted: The country is not supported

Check your PAT's valid country in the **Choose the region / Country** section.

#### Error: The number of API calls has been exceeded

This error occurs when there's an abnormal number of API calls made using the PAT.
The LG ThinQ integration will work properly after some time.

## Debugging

The LG ThinQ integration will show additional information by enabling log configuration. Add the line to your {% term "`configuration.yaml`" %}:

> 

Reviewer: ```suggestion
Don't forget to remove the line again once you are done debugging.
```

> Don't forget to remove the line again once you are done debugging.

Reviewer: ```suggestion
Don't forget to remove the line again once you are done debugging.

```

```yaml
logger:
  default: info
  logs:
    homeassistant.components.lg_thinq: debug

> ```

Reviewer: ```suggestion
```
Don't forget to remove the line again once you are done with debugging.
```

- [Stick Cleaner](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Stick-Cleaner/)
- [Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)
- [System Boiler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/system-boiler/)
- [Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washer/)
- [Washtower](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washtower-single-unit/)
- [Washtower Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washtower-dryer/)
- [Washtower Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washtower-washer/)
- [Water Heater](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-heater/)
- [Water Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-purifier/)
- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/wine-cellar/)
{% note %}
Humidifier, Washcombo Main and Washcombo Mini are also supported, but the links are not available yet.

> {% endnote %}

Author: Updated

- [Stick Cleaner](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Stick-Cleaner/)
- [Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)
- [System Boiler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/System-Boiler/)
- [Washcombo Main](Note: There is no available link yet.)
- [Washcombo Mini](Note: There is no available link yet.)
- [Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washer/)
- [Washtower](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Single-Unit/)
- [Washtower Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Dryer/)
- [Washtower Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Washer/)
- [Water Heater](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Water-Heater/)
- [Water Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-purifier/)
- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/)

> 

Author: Updated

---
title: LG ThinQ
description: Setup for LG ThinQ Integration.
featured: true
ha_category:
  - Hub
ha_release: "2024.10"
ha_iot_class: Cloud Push
ha_code_owners:
  - '@LG-ThinQ-Integration'
ha_config_flow: true
ha_domain: lg_thinq
ha_platforms:
  - switch
ha_dhcp: true
ha_integration_type: integration
---

The **LG ThinQ** integration allows you to connect LG ThinQ devices to Home Assistant. The features of this integration include:

- Control LG appliances as Home Assistant entities through the [LG ThinQ Connect API](https://thinq.developer.lge.com/ko/cloud/).

## Prerequisites

> - This feature works via LG ThinQ cloud. Therefore, an internet connection is required.

Author: Applied.

## Supported Devices

### Appliance

Support LG Appliances as follows:

- [Air Conditioner](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/air-conditioner/)
- [Air Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/air-purifier/)
- [Air Purifier Fan](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/air-purifier-fan/)
- [Ceiling Fan](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/ceiling-fan/)
- [Cooktop](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Cooktop/)
- [Dehumidifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dehumidifier/)
- [Dishwasher](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/dish-washer/)
- [Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Dryer/)
- [Home Brew](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Home-Brew/)
- [Hood](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Hood/)

> - Humidifier

Author: A note is added.

- [Kimchi Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Kimchi-Refrigerator/)
- [Microwave Oven](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Microwave-Oven/)
- [Oven](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Oven/)
- [Plant Cultivator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Plant-Cultivator/)
- [Refrigerator](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/refrigerator/)
- [Robot Cleaner](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/robot-cleaner/)
- [Stick Cleaner](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Stick-Cleaner/)
- [Styler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/styler/)
- [System Boiler](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/System-Boiler/)
- Washcombo Main

> - Washcombo Mini

Author: A note is added.

- [Washtower Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Dryer/)
- [Washtower Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Washer/)
- [Water Heater](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Water-Heater/)
- [Water Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-purifier/)
- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/)

## Troubleshooting

### Setup

#### Aborted: The token is not valid

This error occurs when the Personal Access Token (PAT) is invalid or entered incorrectly. Please visit the [Personal Access Token Page](https://d1jykc6oogauei.cloudfront.net/) page to check if your token is valid.

#### Aborted: The country is not supported

Check your PAT's valid country in the **Choose the region / Country** section.

#### Error: The number of API calls has been exceeded

This error occurs when there's an abnormal number of API calls made using the PAT.

> 

Author: Removed

## Debugging

The LG ThinQ integration will show additional information by enabling log configuration. Add the line to your {% term "`configuration.yaml`" %}:

```yaml
logger:
  default: info
  logs:
    homeassistant.components.lg_thinq: debug

> ```

Reviewer: maybe we should add a step to remove the debug level once they are done debugging?

Author: Do you mean we need to mention how to remove the debug level?
It is just removing the relevant line from the configuration.yaml.

Author: @c0ffeeca7 
How can I resolve this one?
I can't find any other integrations to refer.
We just referred other integrations like smartthings.

- [Washtower Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Dryer/)
- [Washtower Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Washer/)
- [Water Heater](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Water-Heater/)
- [Water Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-purifier/)
- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/)

## Troubleshooting

### Setup

#### Aborted: The token is not valid

This error occurs when the Personal Access Token (PAT) is invalid or entered incorrectly. Please visit the [Personal Access Token Page](https://d1jykc6oogauei.cloudfront.net/) page to check if your token is valid.

#### Aborted: The country is not supported

Check your PAT's valid country in the **Choose the region / Country** section.

#### Error: The number of API calls has been exceeded

This error occurs when there's an abnormal number of API calls made using the PAT.

> 

Reviewer: an recommendation what to do when that happens?

Author: Actually no recommendation but waiting.
Then do we have to remove it?

Reviewer: nono, that's fine. Just tell them to wait then. I think it's better if it's documented, even if it seems obvious.

Author: Added it as below.

The LG ThinQ integration will work properly after some time.

---
title: LG ThinQ
description: Setup for LG ThinQ Integration.
featured: true
ha_category:
  - Hub
ha_release: "2024.10"
ha_iot_class: Cloud Push
ha_code_owners:
  - '@LG-ThinQ-Integration'
ha_config_flow: true
ha_domain: lg_thinq
ha_platforms:
  - switch
ha_dhcp: true
ha_integration_type: integration
---

> The LG ThinQ integration allows you to connect LG ThinQ devices to Home Assistant. The features of this integration include:

Reviewer: ```suggestion
The **LG ThinQ** integration allows you to connect LG ThinQ devices to Home Assistant. The features of this integration include:
```

- Control LG appliances as Home Assistant entities through the [LG ThinQ Connect API](https://thinq.developer.lge.com/ko/cloud/).

> - This feature works via LG ThinQ cloud; therefore, an internet connection is required.

Reviewer: ```suggestion
```
I think this is better placed in the Prereq section

## Prerequisites

> - An internet connection and a [personal access token](https://connect-pat.lgthinq.com) are required to use the [LG ThinQ Connect API](https://thinq.developer.lge.com/ko/cloud/).

Reviewer: ```suggestion
- This feature works via LG ThinQ cloud, therefore an internet connection is required.
- A [personal access token](https://connect-pat.lgthinq.com) to use the [LG ThinQ Connect API](https://thinq.developer.lge.com/ko/cloud/).
```

### Personal Access Token (PAT)

1. Access the **[personal access token](https://connect-pat.lgthinq.com)** page (requires an LG ThinQ account).

> 2. Click the **ADD NEW TOKEN** button.

Reviewer: ```suggestion
2. Select **ADD NEW TOKEN**.
```

3. Enter a new token name and select the following authorized scopes: 

>     - Permission to view all devices
>     - Permission to view all device statuses
>     - All device control rights
>     - All device event subscription rights
>     - All device push notification permissions

Reviewer: ```suggestion
    - **Permission to view all devices**
    - **Permission to view all device statuses**
    - **All device control rights**
    - **All device event subscription rights**
    - **All device push notification permissions**
```

> 4. Click the **CREATE TOKEN** button.

Reviewer: ```suggestion
4. Select **CREATE TOKEN**.
```

5. Once all the steps are completed, you will see that a **PAT** has been generated.

### MQTT

The LG ThinQ integration connects to the LG ThinQ Cloud via MQTT and receives push status updates from the devices. This is a part of the LG ThinQ API implementation and is automatically connected through internal operations, so no additional settings are required.

> {% note %} MQTT connections are only supported in the Appliance feature.{% endnote %}

Reviewer 2: This is not in the current integration

Author: Removed

> {% note %} MQTT connections are only supported in the Appliance feature.{% endnote %}

Reviewer: ```suggestion
{% note %} MQTT connections are only supported in the **Appliance** feature.{% endnote %}
```

> ## Setup
> 
> To set up the integration, please follow these steps:
> 

Reviewer: ```suggestion
```

{% include integrations/config_flow.md %}

1. Enter the information to use LG ThinQ Connect API:
   - The **Token Value** obtained through the PAT issuance process. (Required)
   - The entry name.
2. Choose the region (country).

> ## Delete

Reviewer: ```suggestion
## Removing the LG ThinQ integration
```

> To remove the integration, please follow these steps:

Reviewer: ```suggestion
To remove the LG ThinQ integration, follow these steps:
```

> 1. Go to the Home Assistant Integrations page.

Reviewer: ```suggestion
1. Go to **{% my integrations icon title="Settings > Devices & Services" %}**.
```
There's a my link for this page.

> 2. Select the LG ThinQ integration instance.

Reviewer: ```suggestion
2. Select the LG ThinQ integration.
```

> 3. Click on the options menu and choose Delete.

Reviewer: ```suggestion
3. Select the three dots {% icon "mdi:dots-vertical" %} menu, then select **Delete**.
```

> {% note %} Deleting the integration will remove all entities associated with your LG ThinQ devices' features. {% endnote %}

Reviewer 2: Not sure if this is LG specific

Reviewer: Good point. we should add a general procedure that integration docs can point to, and then add the integration specific information only.

Author: Do you mean we don't need it here because it is not LG specific? 
Then let me remove it.

Author: Removed.

- Washcombo Main
- Washcombo Mini
- [Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/washer/)
- [Washtower](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Single-Unit/)
- [Washtower Dryer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Dryer/)
- [Washtower Washer](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/WashTower-Washer/)
- [Water Heater](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Water-Heater/)
- [Water Purifier](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/water-purifier/)
- [Wine Cellar](https://thinq.developer.lge.com/en/cloud/docs/thinq-connect/device-profile/Wine-Cellar/)

## Troubleshooting

### Setup

#### Aborted: The token is not valid

This error occurs when the Personal Access Token (PAT) is invalid or entered incorrectly. Please visit the [Personal Access Token Page](https://d1jykc6oogauei.cloudfront.net/) page to check if your token is valid.

#### Aborted: The country is not supported

> Check your PAT's valid country in the "Choose the region / Country" section.

Reviewer: ```suggestion
Check your PAT's valid country in the **Choose the region / Country** section.
```
